src/TopSurface.cpp

```diff
@@ -37,14 +37,28 @@ bool TopSurface::ironing(const SliceMeshStorage& mesh, const GCodePathConfig& li
     const bool zig_zaggify_infill = pattern == EFillMethod::ZIG_ZAG;
     constexpr bool connect_polygons = false; // midway connections can make the surface less smooth
     const coord_t line_spacing = mesh.settings.get<coord_t>("ironing_line_spacing");
-    const coord_t outline_offset = -mesh.settings.get<coord_t>("ironing_inset");
     const coord_t line_width = line_config.getLineWidth();
     const std::vector<AngleDegrees>& top_most_skin_angles = (mesh.settings.get<size_t>("roofing_layer_count") > 0) ? mesh.roofing_angles : mesh.skin_angles;
     assert(top_most_skin_angles.size() > 0);
     const AngleDegrees direction = top_most_skin_angles[layer.getLayerNr() % top_most_skin_angles.size()] + AngleDegrees(90.0); //Always perpendicular to the skin lines.
     constexpr coord_t infill_overlap = 0;
     constexpr int infill_multiplier = 1;
     constexpr coord_t shift = 0;
+
+    coord_t ironing_inset = -mesh.settings.get<coord_t>("ironing_inset");
+    if (pattern == EFillMethod::ZIG_ZAG && ironing_inset == 0)
+    {
+        //Compensate for the outline_offset decrease that takes place when using the infill generator to generate ironing with the zigzag pattern
+        const float width_scale = (float)mesh.settings.get<coord_t>("layer_height") / mesh.settings.get<coord_t>("infill_sparse_thickness");
```

Comment:
Infill and the layer height have nothing to do with this, right? This is using the skin line width.

Also, use the Ratio type for this variable.

Reply:
The infill generator uses the width_scale when getting the offset for the zigzag pattern (line 101-105 in TopSurface.cpp https://github.com/Ultimaker/CuraEngine/blob/master/src/infill.cpp#L103). When ironing is done using the infill generator, we must counteract this offset in order to get exactly an ironing inset of 0. This is why width scale is used here.

Reply:
Ah, strange that it does that. The Infill Layer Thickness shouldn't be applicable to skin at all, I'd think. Nor to the parts of infill that are thinner than the Infill Layer Thickness. Fine then, but we might be looking at a bug here or at least some needlessly complex code. But if the infill generator is at fault it could be hard to fix now.